Use cascade-alias instead of related-property in CSSProperties.json

anttijk opened the pull request:
#### ed167d83d75e6576619e1d72c55f61acfd66271e
<pre>
Use cascade-alias instead of related-property in CSSProperties.json
<a href="https://bugs.webkit.org/show_bug.cgi?id=277591">https://bugs.webkit.org/show_bug.cgi?id=277591</a>
<a href="https://rdar.apple.com/133128418">rdar://133128418</a>

Reviewed by Tim Nguyen.

It is a simpler and faster mechanism as it does not turn properties into deferred properties.

box-shadow/-webkit-box-shadow is the only remaining related-property pair. It requires a little bit of
refactoring.

* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:

Aliases are not part of computed style enumeration.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:
(StyleProperty.is_skipped_from_computed_style):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::exposedComputedCSSPropertyIDs):

Decide dynamically if the property is exposed as it may be affected by settings.

* Source/WebCore/rendering/style/FillLayer.cpp:
(WebCore::FillLayer::FillLayer):

The m_compositeSet bit was mistakenly always set for mask layers, preventing composite propagation
to other layers.

Canonical link: <a href="https://commits.webkit.org/281816@main">https://commits.webkit.org/281816@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ae4fdf098f6fff4815a4f2bc6a5ec95b22dd33d2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61110 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40471 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13691 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65051 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11659 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63240 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48148 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11934 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49402 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/8110 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/63144 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37656 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52946 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30232 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/34336 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10171 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10571 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56153 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10466 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66783 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5057 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10280 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56774 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5079 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52909 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56967 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13629 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4188 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36275 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37358 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38452 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37102 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/CSSProperties.json:
```diff
@@ -7414,8 +7408,8 @@
         },
         "-webkit-backdrop-filter": {
             "codegen-properties": {
-                "related-property": "backdrop-filter",
-                "conditional-converter": "FilterOperations",
+                "cascade-alias": "backdrop-filter",
+                "skip-builder": true,
```
Would this break `-webkit-backdrop-filter` on down levels where `cssUnprefixedBackdropFilterEnabled` is turned off?

nt1m replied:
You can provide an array to `codegen-properties` fwiw, with the first item in the array being for `ENABLE_UNPREFIXED_BACKDROP_FILTER` and the other one being for `!ENABLE_UNPREFIXED_BACKDROP_FILTER`. 

See `font-stretch` for instance

anttijk replied:
> Would this break `-webkit-backdrop-filter` on down levels where `cssUnprefixedBackdropFilterEnabled` is turned off?

I don't think so? This patch only affects what happens during cascading, the setting seems to affect parse and compositing time only.

nt1m replied:
I guess it would be weird to alias to a property that's disabled, but I trust your judgment on whether that's fine to do.

anttijk committed:
Use cascade-alias instead of related-property in CSSProperties.json
https://bugs.webkit.org/show_bug.cgi?id=277591
rdar://133128418

Reviewed by Tim Nguyen.

It is a simpler and faster mechanism as it does not turn properties into deferred properties.

box-shadow/-webkit-box-shadow is the only remaining related-property pair. It requires a little bit of
refactoring.

* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:

Aliases are not part of computed style enumeration.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:
(StyleProperty.is_skipped_from_computed_style):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::exposedComputedCSSPropertyIDs):

Decide dynamically if the property is exposed as it may be affected by settings.

* Source/WebCore/rendering/style/FillLayer.cpp:
(WebCore::FillLayer::FillLayer):

The m_compositeSet bit was mistakenly always set for mask layers, preventing composite propagation
to other layers.

Canonical link: https://commits.webkit.org/281816@main
